docs/python/notebooks/onnxruntime-nuphar-tutorial.ipynb

```diff
@@ -0,0 +1,681 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Copyright (c) Microsoft Corporation. All rights reserved.  \n",
+    "Licensed under the MIT License."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# ONNX Runtime: Tutorial for Nuphar execution provider\n",
+    "**Accelerating model inference via compiler, using Docker Images for ONNX Runtime with Nuphar**\n",
+    "\n",
+    "This example shows how to accelerate model inference using Nuphar.\n",
+    "\n",
+    "#### Tutorial Roadmap:\n",
+    "1. Create and run inference on a simple ONNX model, and understand how ***compilation*** works in Nuphar.\n",
+    "2. Create and run inference on a ***LSTM*** model, run symbolic shape inference, edit LSTM ops to Scan, and check Nuphar speedup.\n",
+    "3. ***Quantize*** the LSTM model and check speedup in Nuphar (CPU with AVX2 support is required).\n",
+    "4. Working on a real model: ***Bidirectional Attention Flow ([BiDAF](https://arxiv.org/pdf/1611.01603))*** from onnx model zoo.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Create and run inference on a simple ONNX model\n",
+    "Let's start with a simple model: Y = ((X + X) * X + X) * X + X"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import onnx\n",
+    "from onnx import helper\n",
+    "\n",
+    "model = onnx.ModelProto()\n",
+    "opset = model.opset_import.add()\n",
+    "opset.domain == 'onnx'\n",
+    "opset.version = 7 # ONNX opset 7 is required for LSTM op later\n",
+    "\n",
+    "graph = model.graph\n",
+    "X = 'input'\n",
+    "Y = 'output'\n",
+    "\n",
+    "# declare graph input/ouput with shape [seq, batch, 1024]\n",
+    "dim = 1024\n",
+    "model.graph.input.add().CopyFrom(helper.make_tensor_value_info(X, onnx.TensorProto.FLOAT, ['seq', 'batch', dim]))\n",
+    "model.graph.output.add().CopyFrom(helper.make_tensor_value_info(Y, onnx.TensorProto.FLOAT, ['seq', 'batch', dim]))\n",
+    "\n",
+    "# create nodes: Y = ((X + X) * X + X) * X + X\n",
+    "num_nodes = 5\n",
+    "for i in range(num_nodes):\n",
+    "  model.graph.node.add().CopyFrom(helper.make_node('Mul' if i % 2 else 'Add',\n",
+    "                                                   [X, X if i == 0 else 'out_'+str(i-1)],\n",
+    "                                                   ['out_'+str(i) if i < num_nodes - 1 else Y],\n",
+    "                                                   'node'+str(i)))\n",
+    "\n",
+    "# save the model\n",
+    "simple_model_name = 'simple.onnx'\n",
+    "onnx.save(model, simple_model_name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When loading the model we created above, Nuphar execution provider would be used. We can use settings string to check the generated code.\n",
```

Comment:
Probably modify the sentence to "We will use nuphar execution provider to run the inference for the model that we created above"?